src/uct/ib/dc/dc_mlx5_ep.h

```diff
@@ -434,23 +434,27 @@ static inline ucs_status_t uct_dc_mlx5_iface_dci_get(uct_dc_mlx5_iface_t *iface,
     uct_rc_txqp_t *txqp;
     int16_t available;
 
+    if (!uct_rc_iface_have_tx_cqe_avail(&iface->super.super)) {
```

Comment:
looks like the logic of force signaling when low on QP resources is removed?
BTW do we always have moderation=0 for DC? also for rand policy?

Reply:
you meant CQ resources? Yes, currently we do not use moderation for DC for all policies.

Reply:
ok, so let's force/assert/ensure + document this in the code

Reply:
added assert when checking CQE. We set tx_moderation to 0 in iface init right before setting the DCI policy - so no sense to add assert there.